nixos/modules/services/networking/supplicant.nix

```diff
@@ -44,19 +44,10 @@ let
 
         preStart = ''
           ${optionalString (suppl.configFile.path!=null) ''
-            touch -a ${suppl.configFile.path}
-            chmod 600 ${suppl.configFile.path}
+            (umask 077 && touch -a "${suppl.configFile.path}")
           ''}
           ${optionalString suppl.userControlled.enable ''
-            if ! test -e ${suppl.userControlled.socketDir}; then
-                mkdir -m 0770 -p ${suppl.userControlled.socketDir}
-                chgrp ${suppl.userControlled.group} ${suppl.userControlled.socketDir}
-            fi
-
-            if test "$(stat --printf '%G' ${suppl.userControlled.socketDir})" != "${suppl.userControlled.group}"; then
-                echo "ERROR: bad ownership on ${suppl.userControlled.socketDir}" >&2
-                exit 1
-            fi
```

Comment:
Is there a downside to overwriting the group of an existin directory? Maybe we should keep this but put it before the `install`.

Reply:
> Is there a downside to overwriting the group of an existin directory?

I don't know. I'm not really familiar with wpa_supplicant. I tested if ACLs are preserved with `install`, and they are.

My thinking:

Old code: configure dir once and fail if group ownership differs later.
New code: configure dir always. (The code will only fail if there is insufficient permissions to configure things.)

I don't see any benefit of the old code. The new one automates something the old code would require user action to solve.